batch/batch/server/database.py

```diff
@@ -25,7 +25,18 @@ def __init__(self, db):
         return await super().get_all_records()
 
     async def get_records(self, ids, fields=None):
-        return await super().get_record({'id': ids}, fields)
+        return await super().get_records({'id': ids}, fields)
+
+    async def get_undeleted_records(self, ids, user):
+        async with self._db.pool.acquire() as conn:
+            async with conn.cursor() as cursor:
+                batch_name = self._db.batch.name
+                where_template, where_values = make_where_statement({'id': ids, 'user': user})
+                sql = f"""SELECT * FROM `{self.name}` WHERE {where_template} AND EXISTS 
+                (SELECT id from `{batch_name}` WHERE `{batch_name}`.id = batch_id AND `{batch_name}`.deleted = 0)"""
```

Comment:
Do you want `.deleted = FALSE`?   It seems we're using `BOOLEAN` and `TRUE` elsewhere.  Or just `.deleted` with no comparison, integers appear to get implicitly casted to boolean values as needed.